File: tst/cross.c
#include "pico.h"
#include "../check.h"

int main (void) {
    pico_init(1);
    pico_set_window_title("Hello World!");
    for (int i=0; i<100; i++) {
        pico_output_draw_pixel (
            (Pico_Rel_Pos) { '!', {i, i}, PICO_ANCHOR_C }
        );
        pico_output_draw_pixel (
            (Pico_Rel_Pos) { '!', {99-i, i}, PICO_ANCHOR_C }
        );
        pico_input_delay(5);
        switch (i) {
            case 0:
                _pico_check("cross-01");
                break;
            case 49:
                _pico_check("cross-02");
                break;
            case 99:
                _pico_check("cross-03");
                break;
        }
    }
    pico_init(0);
    return 0;
}

File: lua/tst/cross.lua
require 'pico.check'

--os.execute("mkdir -p /tmp/cross")

pico.init(true)
pico.set.window { title = "Hello World!" }

for i=0, 99 do
    --pico.output.screenshot("window", string.format("/tmp/cross/%02d.png", i))

    pico.output.draw.pixel { '!', x=i,    y=i }
    pico.output.draw.pixel { '!', x=99-i, y=i }
    pico.input.delay(5)
    if i == 0 then
        pico.check("cross-01")
    elseif i == 49 then
        pico.check("cross-02")
    elseif i == 99 then
        pico.check("cross-03")
    end
end

--os.execute("convert -delay 5 -loop 0 /tmp/cross/*.png ../../res/cross.gif")
--os.execute("rm -rf /tmp/cross")

pico.init(false)
